contracts/ElectionPractice.sol

```diff
@@ -0,0 +1,166 @@
+ *      - Only the contract owner can start voting process and call a voting process to the end.
+ *      - Once a voting proces start, no more user can register as candiate.
+ * 
+ * @notice - Search for "TODO" for what you need to implement.
+ *      
+ *
+*/
+
+/**
+ * @title - Events
+ * @dev - Events will serve two purposes
+ *      - 1. Update Election UI via websocket
+ *      - 2. To save data that you want on the blockchain but won't be needed in the contract
+ * 
+ * @notice - do not change the name of event, otherwise the UI will not update automatically
+*/
+contract Events {
+    event voteStart(uint round);
+    event elected(uint round, address candidate, string name, uint vote);
+    event reset(uint round);
+    event registered(uint round, address candidate);
+    event voteCandidate(uint round, address candidate, address voter);
+    event sponsorCandidate(uint round, address candidate, string name, address sponsor, uint amount);
+    event refund(address candidate, string name, uint amount, uint round);
+} 
+ 
+contract Election is Events {
+
+    /**
+     * @notice - This is for functions that can be used/called by the owner.
+     */
+    address public owner;
+    
+    constructor() public {
+        // TODO - Set up owner right after the contract get deployed.
+    }
+    
+    /**
+     * @title isVoting
+     * @dev a mark to determine if 
+     * @notice a public variable that is used for contract and UI update
+     */
+    bool public isVoting = false;
+    
+    function startVoting() public {
+        /**
+         * TODO - to make voting process start
+         * TODO - only contract owner can call this function
+         * TODO - emit event voteStart
+        */
+    }
+    
+
+    /**
+     * @notice - With solidity, it gives you "mapping" as one of the data store mechanism.
+     *         - However, you cannot iterate it like you would with "array"
+     *         - because it uses hash table as its mechanism.
+     *         - In theroy, you can use "array" to store data as you would with "mapping".
+     *         - Within smart contract, each line of running code cost money.
+     *         - In other words, the bigger of an array it gets the higher of gas a user should pay.
+     * 
+     * @dev - In order for user not spending too much gas to interact with the contract.
+     *      - We set four state variables to achieve.
+     */
+
+     /**
+     * @title candidateData
+     * @param uint - election round
+     * @param address - candidate's wallet address
+     * @param Candidate - candidate's election data
+     */
+    mapping(uint => mapping(address => Candidate)) public candidateData;
+     
+     /**
+     * @title candiatesList
+     * @dev - This will store candiates data for each active round, and will be reset whenever an election is over.
+     *      - The reason of this approach is because we are not able to iterate "mapping" 
```

Comment:
The reason of this approach is because we cannot iterate the keys in the mapping

Reply:
ok